app/src/main/java/ch/hikemate/app/ui/auth/SignInWithEmailScreen.kt

```diff
@@ -0,0 +1,148 @@
+
+import android.widget.Toast
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.foundation.text.selection.TextSelectionColors
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import ch.hikemate.app.R
+import ch.hikemate.app.model.authentication.AuthViewModel
+import ch.hikemate.app.ui.components.BackButton
+import ch.hikemate.app.ui.components.BigButton
+import ch.hikemate.app.ui.components.ButtonType
+import ch.hikemate.app.ui.navigation.NavigationActions
+import ch.hikemate.app.ui.navigation.Route
+import ch.hikemate.app.ui.navigation.Screen
+import ch.hikemate.app.ui.theme.primaryColor
+
+object SignInWithEmailScreen {
+  const val TEST_TAG_TITLE = "sign_in_with_email_title"
+  const val TEST_TAG_EMAIL_INPUT = "sign_in_with_email_name_input"
+  const val TEST_TAG_PASSWORD_INPUT = "sign_in_with_email_password_input"
+  const val TEST_TAG_SIGN_IN_BUTTON = "sign_in_with_email_sign_in_button"
+  const val TEST_TAG_GO_TO_SIGN_UP_BUTTON = "sign_in_with_email_go_to_sign_up_button"
+}
+
+/**
+ * A composable that displays the sign in with email screen.
+ *
+ * @param navigationActions The navigation actions.
+ * @param authViewModel The authentication view model.
+ */
+@Composable
+fun SignInWithEmailScreen(navigationActions: NavigationActions, authViewModel: AuthViewModel) {
+  val context = LocalContext.current
+
+  // Define the colors for the input fields
+  val inputColors =
+      OutlinedTextFieldDefaults.colors()
+          .copy(
+              focusedLabelColor = primaryColor,
+              focusedIndicatorColor = primaryColor,
+              cursorColor = primaryColor,
+              textSelectionColors =
+                  TextSelectionColors(
+                      handleColor = primaryColor,
+                      backgroundColor = primaryColor,
+                  ))
```

Comment:
Praise : very good adaptation to both light and dark modes ! Apart from the `BackButton` but that's not part of the PR.